client/post-editor/editor-sidebar/feedback-view.jsx

```diff
@@ -0,0 +1,116 @@
+import FeedbackRequestForm from './feedback-request-form';
+import FeedbackShare from './feedback-share';
+import SidebarFooter from 'layout/sidebar/footer';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getDraftFeedbackShares } from 'state/selectors';
+import { addDraftShare, revokeDraftShare, restoreDraftShare } from 'state/draft-feedback/actions';
+
+// TODO: Find a clearer word than "share" for the owner of feedback and use it in subcomponent and CSS classes
+export class FeedbackView extends Component {
+	static propTypes = {
+		translate: PropTypes.func.isRequired,
+		close: PropTypes.func.isRequired,
+		siteId: PropTypes.number.isRequired,
+		postId: PropTypes.number.isRequired,
+		shares: PropTypes.array.isRequired,
+		addDraftShare: PropTypes.func.isRequired,
+	};
+
+	state = {
+		// The number of FeedbackShare components that are toggled open in the UI
+		totalOpenedShares: 0,
+	};
+
+	onToggleShare = isToggledOpen => {
+		const { totalOpenedShares } = this.state;
+
+		this.setState( {
+			totalOpenedShares: isToggledOpen ? totalOpenedShares + 1 : totalOpenedShares - 1,
+		} );
+	};
+
+	onAddDraftShare = emailAddress => {
+		const { postId, siteId } = this.props;
+		this.props.addDraftShare( siteId, postId, emailAddress );
+	};
+
+	onRevokeShareAccess = emailAddress => {
+		const { postId, siteId } = this.props;
+		this.props.revokeDraftShare( siteId, postId, emailAddress );
+	};
+
+	onRestoreShareAccess = emailAddress => {
+		const { postId, siteId } = this.props;
+		this.props.restoreDraftShare( siteId, postId, emailAddress );
+	};
+
+	render() {
+		const { shares, translate } = this.props;
+		const allSharesClosed = this.state.totalOpenedShares === 0;
+
+		return (
+			<div className="editor-sidebar__view">
+				<FeedbackSidebarHeader closeFeedback={ this.props.close } />
+				{ allSharesClosed &&
+					<div>
+						<div className="editor-sidebar__feedback-header-image-box" />
+						<FeedbackRequestForm requestFeedback={ this.onAddDraftShare } />
+					</div> }
+				{ shares.length > 0 &&
+					<div>
+						<div
+							className={ classNames( {
+								'editor-sidebar__feedback-list-label': true,
+								'is-hidden': ! allSharesClosed,
+							} ) }
+						>
+							{ translate( 'Friends' ) }
+						</div>
+						{ shares.map( share =>
+							<FeedbackShare
+								key={ share.emailAddress }
+								share={ share }
+								onToggle={ this.onToggleShare }
+								onRevokeAccess={ this.onRevokeShareAccess }
+								onRestoreAccess={ this.onRestoreShareAccess }
+							/>,
+						) }
+					</div> }
+				<SidebarFooter />
+			</div>
+		);
+	}
+}
+
+export default connect(
+	state => {
+		const siteId = getSelectedSiteId( state );
+		const postId = getEditorPostId( state );
+		const shares = getDraftFeedbackShares( state, siteId, postId );
+
+		return { siteId, postId, shares };
+	},
+	dispatch =>
```

Comment:
Theres a really nice shorthand that lets you skip `bindActionCreators`. Instead of supplying a function like 

```js
dispatch => bindActionCreators( { creator1, creator1 } )
```

You can just supply an object like this:
```js
{ creator1, creator1 }
```

So in your case it would look something like:
```js
connect(
  mapStateToProps,
  { addDraftShare, revokeDraftShare, restoreDraftShare }
)( localize( FeedbackView ) );
```

[source](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options)